pkg/io/binaryReader.go

```diff
@@ -59,6 +56,38 @@ func (r *BinReader) ReadArray(t interface{}) interface{} {
 	return arr.Interface()
 }
 
+// ReadArrayTo reads array into value which must be
+// a pointer to a slice.
+func (r *BinReader) ReadArrayTo(t interface{}) {
+	value := reflect.ValueOf(t)
+	if value.Kind() != reflect.Ptr || value.Elem().Kind() != reflect.Slice {
+		panic(value.Type().String() + " is not a pointer to a slice")
+	}
+
+	sliceType := value.Elem().Type()
+	checkHasDecodeBinary(sliceType.Elem())
+
+	if r.Err != nil {
```

Comment:
hm. looks a bit strange. Do you expect r.Err to be not nil after some previous actions on lines 62-68? I can't find anything there which could lead to an err. Could you please point out to that?

Reply:
oh. You check that it doesn't have Err from anything which could happen before this func. Ok